### PR TITLE
Fixed card-layout

### LINKED ISSUE #753: Card layout breaks on a long word
## Description

When a card contains a string that is longer than card's width, the card will either hide the overflow, or break the layout

## Reproduction

1. Open cards in the vocabulary storybook.
2. Change the content: add a really long word or a URL without breaks.
4. See error.

## Expectation
The card content should wrap if it doesn't fit 

### From the storybook:
Screenshot 2020-11-01 080449
### From the Vocabulary docs draft site:
Screenshot 2020-11-01 080500

## Additional context
The fix for this is described on[ css-tricks](https://css-tricks.com/snippets/css/prevent-long-urls-from-breaking-out-of-container/)

## Resolution

- [ ] I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by SaravgiYash**
## Fixes

Fixes #753  by @obulat 

## Description

Added overflow-wrap: break-word; word-wrap: break-word; word-break: break-word;

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: packages/vocabulary/src/styles/card.scss [hunk 1]
```diff
@@ -12,6 +12,9 @@ $card-header-shadow: none;
         }
         .card-content {
             padding: $space-big $space-bigger;
+            overflow-wrap: break-word;
+            word-wrap: break-word;
+            word-break: break-word;
```
Comment (dhruvkb) [inline packages/vocabulary/src/styles/card.scss]: This is deprecated now, in favour of `overflow-wrap: break-word`.

File: packages/vocabulary/src/styles/card.scss [hunk 2]
```diff
@@ -12,6 +12,9 @@ $card-header-shadow: none;
         }
         .card-content {
             padding: $space-big $space-bigger;
+            overflow-wrap: break-word;
+            word-wrap: break-word;
```
Comment (dhruvkb) [inline packages/vocabulary/src/styles/card.scss]: This is just an alias for `overflow-wrap`.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Out of the 3 lines, two should be discarded as they are either aliases or deprecated.

File: packages/vocabulary/src/styles/card.scss (hunk 1 above)
Comment (zackkrida) [inline packages/vocabulary/src/styles/card.scss]:
```suggestion
            overflow-wrap: break-word;
```

Comment (zackkrida) [reply packages/vocabulary/src/styles/card.scss]: `overflow-wrap` is the only property needed.

**Review by zackkrida — CHANGES_REQUESTED** [review]

File: packages/vocabulary/src/styles/card.scss (hunk 1 above)
Comment (SaravgiYash) [reply packages/vocabulary/src/styles/card.scss]: @zackkrida overflow-wrap: break-word; doesn't work for horizontal cards
cards

File: packages/vocabulary/src/styles/card.scss (hunk 2 above)
Comment (SaravgiYash) [reply packages/vocabulary/src/styles/card.scss]: @dhruvkb according to this document overflow-wrap doesn't work on some browser.
card

Comment (obulat) [reply packages/vocabulary/src/styles/card.scss]: It was implemented in Firefox v49, so you can leave only one line as Zack suggested. 
Note: Css-tricks, whose article I linked to in the issue, is a great site, but some articles are dated, apparently. It's always useful to check mdn: https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-wrap. They have a table of browser compatibility in the bottom.
Sorry for the incomplete lead, @Yashs911!

Comment (SaravgiYash) [reply packages/vocabulary/src/styles/card.scss]: @obulat But overflow-wrap: break-word; does not work for horizontal cards. So should I use both overflow-wrap: break-word; and word-break: break-word; ?

Edit: I guess using **overflow-wrap: anywhere;** might solve the problem

File: packages/vocabulary/src/styles/card.scss (hunk 1 above)
Comment (SaravgiYash) [reply packages/vocabulary/src/styles/card.scss]: @zackkrida Using **overflow-wrap: anywhere;** solves the problem.

**Review by zackkrida — APPROVED** [review]
👍

**Pull request merged**